zinsults.plugin.zsh

```diff
@@ -36,7 +36,11 @@ function command_not_found_handler {
 	fi
 	if (($#msgs>0));then
 		RANDOM=$(od -vAn -N4 -tu < /dev/urandom)
-		builtin print -P -f 'zsh: %s\n' "$msgs[RANDOM % $#msgs + 1]"
+		if [[ -w $TTY ]]; then
+			builtin print -P -f 'zsh: %s\n' "$msgs[RANDOM % $#msgs + 1]" >$TTY
```

Comment:
The Ideal solution would be being silent if output is not a tty, so we can just return if output is a pipe or a subshell.

Reply:
Or we can write to stderr (`>&2`), in that case, we won't have to check `$TTY`.